src/prefect_server/api/flows.py

```diff
@@ -355,9 +356,24 @@ async def archive_flow(flow_id: str) -> bool:
 
     # delete scheduled flow runs
     await models.FlowRun.where(
-        {"flow_id": {"_eq": flow_id}, "state": {"_eq": "Scheduled"}}
+        {
+            "flow_id": {"_eq": flow_id},
+            "state": {"_eq": "Scheduled"},
+            "auto_scheduled": {"_eq": True},
+        }
     ).delete()
 
+    # cancel adhoc scheduled flow runs
+    fr_ids = await models.FlowRun.where(
+        {
+            "flow_id": {"_eq": flow_id},
+            "state": {"_eq": "Scheduled"},
+            "auto_scheduled": {"_eq": False},
+        }
+    ).get({"id"})
+    msg = f"Flow {flow_id} was archived."
+    for flow_run in fr_ids:
+        await api.states.cancel_flow_run(flow_run.id, state_message=msg)
```

Comment:
Would it make sense to do a `gather` instead in case one fails?

Reply:
Yup, that's a great idea - will update

Reply:
@madkinsz updated 👍